### PR TITLE
Rewrite of Git module

### PR DESCRIPTION
**Pull request by NicoleRauch**
* Now the async call to git has a callback (previously it was an async call that was executed inline! That's why the SoCraTes wiki notification mail would sporadically fail)
* Now all checks (of repo and git binary) are done when the application starts

Please run some manual smoketests if possible (I did that, too) - maybe even on Windows...?

### CODE REVIEW

File: softwerkskammer/lib/wiki/gitmech.js [hunk 1]
```diff
@@ -62,12 +56,10 @@ module.exports = {
   latestChanges: function (path, moment, callback) {
     gitExec.command(['log', '--since="' + moment.format('MM/DD/YYYY hh:mm:ss') + '"', '--pretty=format:%h%n%H%n%an%n%ai%n%s', '--', path], function (err, data) {
       if (err) { return callback(err); }
-      var logdata = data ? data.toString().split('\n') : [];
-      var group;
+      var logdata = data ? data.split('\n') : [];
       var metadata = [];
```
Comment (leider) [inline softwerkskammer/lib/wiki/gitmech.js]: most `var` declarations could also be replaced by `const`
should we do that?

File: softwerkskammer/lib/wiki/gitmech.js [hunk 2]
```diff
@@ -91,7 +83,7 @@ module.exports = {
   mv: function (oldpath, newpath, message, author, callback) {
     gitExec.command(['mv', oldpath, newpath], function (err) {
       if (err) { return callback(err); }
-      gitExec.command(['commit', '--author="' + author + '"', '-m', message], function (err1) { callback(err1); });
+      commit('', message, author, callback);
```
Comment (leider) [inline softwerkskammer/lib/wiki/gitmech.js]: nice

File: softwerkskammer/lib/wiki/gitmech.js (hunk 1 above)
Comment (NicoleRauch) [inline softwerkskammer/lib/wiki/gitmech.js]: good point, will do

File: softwerkskammer/lib/wiki/gitmech.js (hunk 2 above)
Comment (NicoleRauch) [inline softwerkskammer/lib/wiki/gitmech.js]: B-)

**Pull request merged**